x/observer/keeper/msg_server_update_observer.go

```diff
@@ -0,0 +1,100 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/zeta-chain/zetacore/x/observer/types"
+)
+
+func (k msgServer) UpdateObserver(goCtx context.Context, msg *types.MsgUpdateObserver) (*types.MsgUpdateObserverResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	chains := k.GetParams(ctx).GetSupportedChains()
+	for _, chain := range chains {
+		if !k.IsObserverPresentInMappers(ctx, msg.OldObserverAddress, chain) {
+			return nil, errorsmod.Wrap(types.ErrNotAuthorized, fmt.Sprintf("Observer address is not authorized for chain : %s", chain.String()))
+		}
+	}
+
+	err := k.IsValidator(ctx, msg.NewObserverAddress)
+	if err != nil {
+		return nil, errorsmod.Wrap(types.ErrUpdateObserver, err.Error())
+	}
+
+	ok, err := k.CheckUpdateReason(ctx, msg)
+	if err != nil {
+		return nil, errorsmod.Wrap(types.ErrUpdateObserver, err.Error())
+	}
+	if !ok {
+		return nil, errorsmod.Wrap(types.ErrUpdateObserver, fmt.Sprintf("Unable to update observer with update reason : %s", msg.UpdateReason))
+	}
+
+	// Update all mappers so that ballots can be created for the new observer address
+	k.UpdateObserverAddress(ctx, msg.OldObserverAddress, msg.NewObserverAddress)
+
+	// Update the node account with the new operator address
+	nodeAccount, found := k.GetNodeAccount(ctx, msg.OldObserverAddress)
+	if !found {
+		return nil, errorsmod.Wrap(types.ErrNodeAccountNotFound, fmt.Sprintf("Observer node account not found : %s", msg.Creator))
```

Comment:
We get the node account with `msg.OldObserverAddress` but return an error with `msg.Creator`, is it always the same?

Reply:
Yeah the print statement is wrong